server/src/main/java/io/crate/execution/engine/aggregation/impl/NumericStandardDeviationAggregation.java

```diff
@@ -0,0 +1,209 @@
+                     V state,
+                     Input<?>... args) throws CircuitBreakingException {
+        if (state != null) {
+            BigDecimal value = (BigDecimal) args[0].value();
+            if (value != null) {
+                long sizeBefore = state.size();
+                state.increment(value);
+                long sizeAfter = state.size();
+                ramAccounting.addBytes(sizeBefore - sizeAfter);
+            }
+        }
+        return state;
+    }
+
+    @Override
+    public V reduce(RamAccounting ramAccounting, V state1, V state2) {
+        if (state1 == null) {
+            return state2;
+        }
+        if (state2 == null) {
+            return state1;
+        }
+        state1.merge(state2);
+        return state1;
+    }
+
+    @Override
+    public boolean isRemovableCumulative() {
+        return true;
+    }
+
+    @Override
+    public V removeFromAggregatedState(RamAccounting ramAccounting, V previousAggState, Input<?>[]stateToRemove) {
+        if (previousAggState != null) {
+            BigDecimal value = (BigDecimal) stateToRemove[0].value();
+            if (value != null) {
+                long sizeBefore = previousAggState.size();
+                previousAggState.decrement(value);
+                long sizeAfter = previousAggState.size();
+                ramAccounting.addBytes(sizeBefore - sizeAfter);
+            }
+        }
+        return previousAggState;
+    }
+
+    @Override
+    public BigDecimal terminatePartial(RamAccounting ramAccounting, V state) {
+        return state.result();
+    }
+
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       Version shardCreatedVersion,
+                                                       List<Literal<?>> optionalParams) {
+        Reference reference = aggregationReferences.get(0);
+        if (reference == null) {
+            return null;
+        }
+        if (!reference.hasDocValues()) {
+            return null;
+        }
+
+        NumericType numericType = (NumericType) reference.valueType();
+        Integer precision = numericType.numericPrecision();
+        Integer scale = numericType.scale();
+        if (precision == null || scale == null) {
+            throw new UnsupportedOperationException(
+                    "NUMERIC type requires precision and scale to support aggregation");
+        }
+        if (precision <= NumericStorage.COMPACT_PRECISION) {
+            return new SortedNumericDocValueAggregator<>(
+                    reference.storageIdent(),
+                    (ramAccounting, memoryManager, version) ->
+                        newState(ramAccounting, version, Version.V_6_0_0, memoryManager),
```

Comment:
This looks a bit odd as the 2nd version here is hardcoded, such the check you've introduced will never fail, right?
Honestly, the api (usage) looks odd in general, as the first version to the newState() api should be the creation version of the actual index while the DocValueAggregator is missing this argument. As a followup we may either remove this version arg if not needed anymore or enhance the DocValueAggregator iface to also consume it.

Passing in the `minNodeVersion` as the `indexCreatedVersion` is wrong but without an API change, we cannot do anything better. And I think the `indexCreatedVersion` is not of any interest here?
```suggestion
                        newState(ramAccounting, version, version, memoryManager),
```

Reply:
Opened a PR to address this: https://github.com/crate/crate/pull/17931, so I can rebase this PR on top of the removal of indexVersionCreated.